### PR TITLE
Lightbox

### PR DESCRIPTION
**Pull request by MrJons**
@earleskinner

I still have to write the tests for the Javascript, however thought i'd submit a pull request so review work can be started. 

Note that as this has been rebased on top of [ALP-1764](https://github.com/AVADOLearning/moodle-local_signin/tree/ALP-1764) there will be duplicate files that wont require review here.

Files to be reviewed are: 
- lightbox.php
- templates/lightbox.mustache
- amd/src/lightbox.js

### CODE REVIEW

**Comment by MrJons** [comment]
@earleskinner Behat tests now added and ready for review.

File: classes/form/password_form.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        $pass_attributes = array('placeholder'   => $this->lang_string('form_password_placeholder'),
+                                 'additionalcss' => $this->lang_string('form_password_button_class'),
+                                 'autofocus'     => '');
+        $mform->addElement('password', 'password', $this->lang_string('form_password_label'), $pass_attributes);
+        $mform->setType('password', PARAM_RAW);
+
+        $submit_attributes = array('additionalcss' => $this->lang_string('form_password_button_class'));
+        $mform->addElement('submit', 'submitpassword', $this->lang_string('form_password_button_label'), $submit_attributes);
+
+        $mform->addElement('html', sprintf('<div class="%s"><a href="%s">%s</a></div>',
```
[the remark this answers is not in the document]
Comment (MrJons) [reply classes/form/password_form.php]: I have not actioned this as I didn't construct the forms, I have told Stefan about your recommendation.

File: amd/src/lightbox.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                methodname: WEB_SERVICE_METHOD_NAME,
+                args: {
+                    username: username
+                }
+            }
+        ]);
+    }
+
+    function handleDomainRedirect(response, username, $options) {
+        var currentDomain = new URL(M.cfg.wwwroot);
```
Comment (LukeCarrier) [inline amd/src/lightbox.js]: Since this is a URL rather than just the domain portion, could we rename to `currentUrl`?

Comment (MrJons) [reply amd/src/lightbox.js]: Done. Now on Stefans branch. Will be opening another PR with all the work together.

**Pull request closed without merging**